server/src/main/java/com/cloud/network/vpn/Site2SiteVpnManagerImpl.java

```diff
@@ -561,12 +561,12 @@ public Site2SiteVpnConnection resetVpnConnection(ResetVpnConnectionCmd cmd) thro
         }
         _accountMgr.checkAccess(caller, null, false, conn);
 
-        if (conn.getState() == State.Pending) {
-            conn.setState(State.Disconnected);
-        }
-        if (conn.getState() == State.Connected || conn.getState() == State.Error || conn.getState() == State.Disconnected) {
-            stopVpnConnection(id);
-        }
+        // Set vpn state to disconnected
+        conn.setState(State.Disconnected);
+        _vpnConnectionDao.persist(conn);
```

Comment:
why not just persist DB only when state is pending ?

```
        if (conn.getState() == State.Pending) {
            conn.setState(State.Disconnected);
            _vpnConnectionDao.persist(conn);
        }
```

Reply:
@weizhouapache since we are restting the connection, it should transit to disconnect state irrecpective of the previous state. since we are stopping and starting the vpn connection, checking for pending state or any other state doesnt matter